h2o-core/src/main/java/water/parser/CsvParser.java

```diff
@@ -458,16 +458,24 @@
     return dout;
   }
 
-  @Override protected int fileHasHeader(byte[] bits, ParseSetup ps) {
+  @Override
+  protected int fileHasHeader(byte[] bits, ParseSetup parseSetup) {
     boolean hasHdr = true;
     String[] lines = getFirstLines(bits);
-    if (lines != null && lines.length > 0) {
-      String[] firstLine = determineTokens(lines[0], _setup._separator, _setup._single_quotes);
-      if (_setup._column_names != null) {
-        for (int i = 0; hasHdr && i < firstLine.length; ++i)
+    if (lines != null && lines.length > 1) {
+      String[] firstLine = determineTokens(lines[0], parseSetup._separator, parseSetup._single_quotes);
+      String[] secondLine = determineTokens(lines[1], parseSetup._separator, parseSetup._single_quotes);
```

Comment:
This will not work sometimes because of values that can span multiple lines.

Reply:
That is a good point. However, does the rest of the code count on that as well ?